File: store-micro-service/src/main/java/com/mvqa/marketmicroservice/model/Card.java
package com.mvqa.marketmicroservice.model;


import jakarta.persistence.*;

@Entity
@Table(name = "card")
public class Card {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "description")
    private String description;

    @Column(name = "name")
    private String name;

    @Column(name = "family_name")
    private String familyName;

    @Column(name = "hp")
    private Double hp;

    @Column(name = "attack")
    private Double attack;

    @Column(name = "energy")
    private Double energy;

    @Column(name = "defense")
    private Double defense;

    @Column(name = "actual_value")
    private Double actualValue;

    @Column(name = "image_url")
    private String imageUrl;


    public Card() { }


    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Long getId() {
        return id;
    }

    public Card setId(Long id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Card setName(String name) {
        this.name = name;
        return this;
    }

    public Double getHp() {
        return hp;
    }

    public Card setHp(Double hp) {
        this.hp = hp;
        return this;
    }

    public Double getAttack() {
        return attack;
    }

    public Card setAttack(Double attack) {
        this.attack = attack;
        return this;
    }

    public Double getDefense() {
        return defense;
    }

    public Card setDefense(Double defense) {
        this.defense = defense;
        return this;
    }

    public Double getActualValue() {
        return actualValue;
    }

    public Card setActualValue(Double actualValue) {
        this.actualValue = actualValue;
        return this;
    }

    public String getFamilyName() {
        return familyName;
    }

    public void setFamilyName(String familyName) {
        this.familyName = familyName;
    }

    public Double getEnergy() {
        return energy;
    }

    public void setEnergy(Double energy) {
        this.energy = energy;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: game-back/src/business/Card.ts
export default class Card {
    constructor(id: number) {
        this._id = id;
    }

    private _id: number;

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    private _name: string;

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    private _description: string;

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    private _hp: number;

    get hp(): number {
        return this._hp;
    }

    set hp(value: number) {
        this._hp = value;
    }

    private _energy: number;

    get energy(): number {
        return this._energy;
    }

    set energy(value: number) {
        this._energy = value;
    }

    private _attack: number;

    get attack(): number {
        return this._attack;
    }

    set attack(value: number) {
        this._attack = value;
    }
}